### PR TITLE
ddl: fix ErrGCTooEarly when adding an index to the partition table

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve? 
The partition table has several `PartitionDefinition`. Every `PartitionDefinition` we will read the corresponding columns records and backfill the index data when adding an index. We will deal with `PartitionDefinition` one by one.
Before a `PartitionDefinition` read the records, we will use `reorg.Job.SnapshotVer` to get range information. When we have two `PartitionDefinition`, and the first partition we do a long time. Then fetching the ranges for the second partition may encounter an `ErrGCTooEarly` error. 

### What is changed and how it works?
Using the current version to get the range of a partition table.

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch

Release note

 - Fix the error of `ErrGCTooEarly` when adding an index to the partition table

### CODE REVIEW

File: ddl/index.go [hunk 1]
```diff
@@ -1425,7 +1426,20 @@ func (w *worker) updateReorgInfo(t table.PartitionedTable, reorg *reorgInfo) (bo
...
 	}
 
-	start, end, err := getTableRange(reorg.d, t.GetPartition(pid), reorg.Job.SnapshotVer, reorg.Job.Priority)
+	failpoint.Inject("mockUpdateCachedSafePoint", func(val failpoint.Value) {
+		if val.(bool) {
+			// 18 is physicalShiftBits.
+			ts := oracle.GetPhysical(time.Now()) << 18
+			s := reorg.d.store.(tikv.Storage)
+			s.UpdateSPCache(uint64(ts), time.Now())
+			time.Sleep(time.Millisecond * 3)
```
Comment (wjhuang2016) [inline ddl/index.go]: > the first partition we do a long time 

3 Millisecond is long enough?

**Comment by wjhuang2016** [comment]
> Using the current version to get the range of a partition table.

Why we use `SnapshotVer` before? Is it safe to use the current version here?

File: ddl/index.go (hunk 1 above)
Comment (zimulala) [reply ddl/index.go]: It seems enough.

**Comment by zimulala** [comment]
@wjhuang2016 
This is theoretically possible, but may lead to "ErrGCTooEarly". It is safe to use the current version, and currently adding indexes on non-partitioned also uses this method.

**Comment by zimulala** [comment]
PTAL @wjhuang2016 @crazycs520 @AilinKid @bb7133

File: ddl/index.go [hunk 2]
```diff
@@ -1425,7 +1426,20 @@ func (w *worker) updateReorgInfo(t table.PartitionedTable, reorg *reorgInfo) (bo
 		return true, nil
 	}
 
-	start, end, err := getTableRange(reorg.d, t.GetPartition(pid), reorg.Job.SnapshotVer, reorg.Job.Priority)
+	failpoint.Inject("mockUpdateCachedSafePoint", func(val failpoint.Value) {
+		if val.(bool) {
+			// 18 is physicalShiftBits.
```
Comment (AilinKid) [inline ddl/index.go]: 18 is logicalShiftBits？

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**